src/qt_gui/game_grid_frame.cpp

```diff
@@ -0,0 +1,157 @@
+// SPDX-FileCopyrightText: Copyright 2024 shadPS4 Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "game_grid_frame.h"
+
+GameGridFrame::GameGridFrame(std::shared_ptr<GameInfoClass> game_info_get,
+                             std::shared_ptr<GuiSettings> m_gui_settings, QWidget* parent)
+    : QTableWidget(parent) {
+    m_game_info = game_info_get;
+    m_gui_settings_ = m_gui_settings;
+    icon_size = m_gui_settings->GetValue(gui::m_icon_size_grid).toInt();
+    windowWidth = parent->width();
+    this->setShowGrid(false);
+    this->setEditTriggers(QAbstractItemView::NoEditTriggers);
+    this->setSelectionBehavior(QAbstractItemView::SelectItems);
+    this->setSelectionMode(QAbstractItemView::SingleSelection);
+    this->setVerticalScrollMode(QAbstractItemView::ScrollPerPixel);
+    this->setHorizontalScrollMode(QAbstractItemView::ScrollPerPixel);
+    this->verticalScrollBar()->installEventFilter(this);
+    this->verticalScrollBar()->setSingleStep(20);
+    this->horizontalScrollBar()->setSingleStep(20);
+    this->horizontalHeader()->setVisible(false);
+    this->verticalHeader()->setVisible(false);
+    this->setContextMenuPolicy(Qt::CustomContextMenu);
+    PopulateGameGrid(m_game_info->m_games, false);
+
+    connect(this, &QTableWidget::cellClicked, this, &GameGridFrame::SetGridBackgroundImage);
+
+    connect(this->verticalScrollBar(), &QScrollBar::valueChanged, this,
+            &GameGridFrame::RefreshGridBackgroundImage);
+    connect(this->horizontalScrollBar(), &QScrollBar::valueChanged, this,
+            &GameGridFrame::RefreshGridBackgroundImage);
+    connect(this, &QTableWidget::customContextMenuRequested, this, [=, this](const QPoint& pos) {
+        m_gui_context_menus.RequestGameMenu(pos, m_game_info->m_games, this, false);
+    });
+}
+
+void GameGridFrame::PopulateGameGrid(QVector<GameInfo> m_games_search, bool fromSearch) {
+    QVector<GameInfo> m_games_;
+    this->clearContents();
+    if (fromSearch)
+        m_games_ = m_games_search;
+    else
+        m_games_ = m_game_info->m_games;
+    m_games_shared = std::make_shared<QVector<GameInfo>>(m_games_);
+
+    icon_size = m_gui_settings_->GetValue(gui::m_icon_size_grid)
+                    .toInt(); // update icon size for resize event.
+
+    int gamesPerRow = windowWidth / (icon_size + 20); // 2 x cell widget border size.
+    int row = 0;
+    int gameCounter = 0;
+    int rowCount = m_games_.size() / gamesPerRow;
+    if (m_games_.size() % gamesPerRow != 0) {
+        rowCount += 1; // Add an extra row for the remainder
+    }
+    std::vector<int> indices;
+    for (int i = 0; i < m_games_.size(); i++) {
+        indices.push_back(i);
+    }
+    std::vector<std::future<QPixmap>> futures;
+    for (int index : indices) {
```

Comment:
wouldn't a simple `for (int index = 0; index < m_games_.size(); i++) {` work?

Reply:
Thanks! we can even remove it and move it inside the other loop

Reply:
You're welcome :)
BTW out of curiosity was this function written by AI?